esp-config/src/lib.rs

```diff
@@ -42,3 +40,64 @@ macro_rules! esp_config_bool {
         }
     };
 }
+
+#[macro_export]
+#[doc(hidden)] // to avoid confusion with esp_config_int, let's hide this
+/// Parse a string like "777" into an integer, which _can_ be used in a `const`
+/// context
+macro_rules! esp_config_int_parse {
+    ($ty:ty, $bytes:expr) => {{
+        #[allow(clippy::assign_op_pattern)]
+        let mut bytes = $bytes;
+        let mut val: $ty = 0;
+        let mut sign_seen = false;
+        let mut is_negative = false;
+        while let [byte, rest @ ..] = bytes {
+            match *byte {
+                b'0'..=b'9' => {
+                    val = val * 10 + (*byte - b'0') as $ty;
+                }
+                b'-' | b'+' if !sign_seen => {
+                    if *byte == b'-' {
+                        is_negative = true;
+                    }
+                    sign_seen = true;
+                }
+                _ => ::core::panic!("invalid digit"),
+            }
+            bytes = rest;
+        }
+        if is_negative {
+            let original = val;
+            // subtract twice to get the negative
+            val = val - original;
```

Comment:
I assume `val -= original` isn't working in const context?

Reply:
Yeah, it desugars to the `AddAssign` trait and you can't call trait methods in const context yet

Reply:
^^ yeah no, I should have tried it it works :see_no_evil:. Thanks @bugadani